chore: update dependencies (oct 2025)

dkoo opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

See https://github.com/Automattic/newspack-scripts/pull/219.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo committed:
chore: update dependencies (oct 2025)

dkoo committed:
chore: fix JS for linters

dkoo committed:
chore: remove --legacy-peer-deps

laurelfulford commented:
- [x] Run `npm start` or `npm ci` (no --legacy-peer-deps!)
- [x] Confirm that the install completes successfully
- [x] Run `npm run build` and `npm run watch` and confirm that the assets build successfully
- [x] Run `npm run lint` and confirm that SCSS and JS linting works
- [x] Run `npm run test` and confirm that JS unit tests work (only applicable if the repo has any) -- returns `No JS unit tests in this repository.`
- [x] Run `npm run semantic-release --dry-run` and confirm there's successful output
- [x] Smoke test both WP admin and front-end functionality—there should be no significant changes

laurelfulford commented:
Lint is working great 😆 

```
src/content-distribution/content-distribution-panel/style.scss
  10:30  ✖  Unexpected whitespace after "("               @stylistic/selector-pseudo-class-parentheses-space-inside
  10:66  ✖  Unexpected whitespace before ")"              @stylistic/selector-pseudo-class-parentheses-space-inside
  15:3   ✖  Expected empty line before rule               rule-empty-line-before
  28:3   ✖  Expected empty line before rule               rule-empty-line-before
  31:3   ✖  Expected empty line before rule               rule-empty-line-before
  38:3   ✖  Expected empty line before rule               rule-empty-line-before
  41:3   ✖  Expected empty line before rule               rule-empty-line-before
  46:3   ✖  Expected empty line before rule               rule-empty-line-before
  46:9   ✖  Unexpected whitespace after "("               @stylistic/selector-pseudo-class-parentheses-space-inside
  46:24  ✖  Unexpected whitespace before ")"              @stylistic/selector-pseudo-class-parentheses-space-inside
  49:3   ✖  Expected empty line before rule               rule-empty-line-before
  49:9   ✖  Unexpected whitespace after "("               @stylistic/selector-pseudo-class-parentheses-space-inside
  49:34  ✖  Unexpected whitespace before ")"              @stylistic/selector-pseudo-class-parentheses-space-inside
  62:2   ✖  Expected empty line before rule               rule-empty-line-before
  63:3   ✖  Expected empty line before rule               rule-empty-line-before
  64:10  ✖  Expected "currentColor" to be "currentcolor"  value-keyword-case

src/content-distribution/incoming-post/style.scss
   3:2    ✖  Expected empty line before rule                                rule-empty-line-before
  23:3    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  26:4    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  30:22   ✖  Selector should use lowercase and separate words with hyphens  selector-id-pattern
  30:100  ✖  Selector should use lowercase and separate words with hyphens  selector-id-pattern
  33:4    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  36:5    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  41:104  ✖  Expected line length to be no more than 80 characters          @stylistic/max-line-length
  42:2    ✖  Selector should use lowercase and separate words with hyphens  selector-id-pattern
  43:3    ✖  Expected empty line before rule                                rule-empty-line-before
  47:28   ✖  Unexpected whitespace after "("                                @stylistic/selector-pseudo-class-parentheses-space-inside
  47:62   ✖  Unexpected whitespace before ")"                               @stylistic/selector-pseudo-class-parentheses-space-inside
  48:4    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  51:5    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  58:3    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  61:4    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  65:84   ✖  Expected line length to be no more than 80 characters          @stylistic/max-line-length
  67:3    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
  70:4    ✖  Expected empty line before at-rule                             at-rule-empty-line-before
```

laurelfulford reviewed (changes requested):
When I got to edit posts on this PR, I get this error on both the Hub and Network; I can't recreate it on trunk:

<img width="1174" height="133" alt="CleanShot 2025-10-31 at 13 05 18" src="https://github.com/user-attachments/assets/3ddd3716-4f05-4c44-8bca-4d9e93b0cbfb" />

In the inspector, it comes up as `Uncaught TypeError: _isSavingPost is not a function`.

I didn't run into errors on the other screens - just let me know if you can't recreate this!

Edited to add: I suspect the same thing would happen with _isCleanNewPost since they're formatted the same in the PR, but I didn't actually see the same error for that one!

dkoo committed:
chore: fix SCSS for linters

dkoo committed:
chore: add .stylelintrc file

dkoo commented:
@laurelfulford whoops, it looks like this repo was lacking a .stylelintrc.js file because it only recently started containing any SCSS. I've added that and ran the `npm run format:scss` to fix most of the linting errors.

I also saw that there's some redundancy in the `.stylelintrc.js` files across our repos. I moved the `ignoreFiles` rules from each individual repo's config to [`newspack-scripts/config/stylelint.config.js`](https://github.com/Automattic/newspack-scripts/pull/219/commits/ce6f84b361f69ba99904d11d14ed394bb82a58ed). Some repos were also missing a `.stylelintrc.js` so I added one where needed.

You may need to remove `node_modules` and run `npm ci` again before `lint:scss` will pass, but you can also just check whether the job passes below 👇

dkoo committed:
chore: add lint-js-scss job to CI config

laurelfulford commented:
Thanks @dkoo! I'm still getting a couple lint errors locally (weird though, your commit is happy!):

```
release/newspack-network/src/content-distribution/content-distribution-panel/style.scss
  64:10  ✖  Expected "currentColor" to be "currentcolor"  value-keyword-case

release/newspack-network/src/content-distribution/incoming-post/style.scss
  30:22   ✖  Selector should use lowercase and separate words with hyphens  selector-id-pattern
  30:100  ✖  Selector should use lowercase and separate words with hyphens  selector-id-pattern
  42:2    ✖  Selector should use lowercase and separate words with hyphens  selector-id-pattern
```

I think as long as the tests are passing in GitHub we could tackle these as a separate PR 🙂 I think the main remaining issue is that [`Uncaught TypeError: _isSavingPost is not a function.`](https://github.com/Automattic/newspack-network/pull/282#pullrequestreview-3406191708) error.

dkoo committed:
chore: fix function names

dkoo commented:
Whoops, missed that comment. 10201a2 should fix the JS error!

laurelfulford reviewed (approved):
Thanks @dkoo! That took care of the error 🎉

dkoo committed:
chore: bump newspack packages
